### PR TITLE
docs: Format and improve descriptions of the public API's class, methods, and parameters. (EDU-2483)

### PR DESCRIPTION
**Pull request by jamesneimanconsulting**
docs: Format and improve descriptions of the public API's class, methods, and parameters.
Provide additional detail to the descriptions and consistency with readme.io.

### CODE REVIEW

**Comment by jamesneimanconsulting** [comment]
@jonslaught , @mikeng13 Kindly review and merge if approved.

File: lib/optimizely.rb [hunk 1]
```diff
@@ -84,20 +84,20 @@ def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = n
...
 
-    # Activates an A/B test for a user, deciding whether they qualify for the experiment, bucketing them
-    # into a variation if they do, and sending an impression event to Optimizely.
+    # Activates an A/B test for a user, determines whether they qualify for the experiment, buckets a qualified
+    # user into a variation, and sends an impression event to Optimizely.
     #
     # This method takes into account the user `attributes` passed in, to determine if the user
     # is part of the audience that qualifies for the experiment.
     #
     # For more information, see https://github.com/optimizely/ruby-sdk/blob/master/lib/optimizely.rb.
```
Comment (jonslaught) [inline lib/optimizely.rb]: Does this file just link to itself? What's the point?

File: lib/optimizely.rb [hunk 2]
```diff
@@ -84,20 +84,20 @@ def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = n
...
     #
-    # @param experiment_key [string]  The key of the experiment for which to activate the variation.
-    # @param user_id        [string]  The ID of the user for whom to activate the variation.
+    # @param experiment_key [string]  The key of the variation's experiment to activate.
```
Comment (jonslaught) [inline lib/optimizely.rb]: This shoud just hbe "The experiment to activate". It doesn't make sense to refer to the "variation's experiment".

File: lib/optimizely.rb (hunk 1 above)
Comment (jonslaught) [reply lib/optimizely.rb]: Shouldn't it link to our readme API reference instead?

File: lib/optimizely.rb [hunk 3]
```diff
@@ -338,7 +335,7 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
     # 
     # For more information, see https://docs.developers.optimizely.com/full-stack/docs/get-enabled-features.
     #
-    # @param user_id    [string]             The ID of the participant in the experiment.
+    # @param user_id    [string]             The ID of the user who may have features enabled in one or more experiments.
```
Comment (jonslaught) [inline lib/optimizely.rb]: "The ID of the user to check" would be simpler here.

**Comment by jamesneimanconsulting** [comment]
@jonslaught I've made the changes to the branch.

**Comment by jamesneimanconsulting** [comment]
@jonslaught Thanks. I've updated all 5 repos now.

**Comment by jamesneimanconsulting** [comment]
@jonslaught , @mikeng13 I've removed the trailing white spaces on the various lines.

**Comment by mikeng13** [comment]
This is outdated. Will close and follow-up on a different PR.

**Pull request closed without merging**